### PR TITLE
Serve iframe from django + resurrect iodide local-only dev environment

### PR DESCRIPTION
**Pull request by wlach**
* The iframe is now served via django from the special `/notebooks/eval-frame/` endpoint. This is architecturally a little cleaner and paves the way towards #1943
* For testing changes to the editor only, you can now use the (renamed)
"simple-serve" command.

### Pull Request checklist

- [x] **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [x] **Changelog**: This PR updates the [changelog](../CHANGELOG.md) with any user-visible changes.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not

### CODE REVIEW

**Comment by wlach** [comment]
@bcolloran please review js parts (python portions optional)
@jezdez could you review the python changes? (and their interactions with the js parts, if you feel up to it). likely a prereq to reviewing is understand the "iodide content in an iframe model" that we have -- either me or @bcolloran could explain this. maybe we could go over it in the iodide meeting tomorrow if you'll be around
@robotblake this is more an FYI than anything else, but grepping through the PR for "middleware" might highlight areas of concern from a deployment POV

File: docs/local-dev.md [hunk 1]
```diff
@@ -8,26 +8,26 @@ Run `npm install` after cloning this repository.
...
+The command runs in watch mode, so changes to files will be detected and bundled automatically, but you will need to refresh the page in your browser manually to see the changes -- we have disabled "hot reloading" because automatically refreshing the browser would cause any active notebooks to lose their evaluation state.
 
-If you require verbose Redux logging, you can use the command `npm run start-and-serve -- reduxVerbose`
-
-#### Exporting from client-only dev mode
-
-In this mode, resource paths are set to be relative to the `dev/` directory. Thus, if you export a bundled notebook from a dev notebook, you need to be sure save the exported HTML file in the `dev/` folder for the relative paths to correctly resolve the required js, css, and font files (and if you want to share a notebook that you created in a dev environment, you'll need to update the paths to point to the web-accessible resources at `iodide.io` and `iodide.app`).
+If you require verbose Redux logging, you can use the command `npm run simple-serve -- reduxVerbose`
```
Comment (bcolloran) [inline docs/local-dev.md]: oops, `--reduxVerbose` doesn't work anymore, one needs to set the environment var `REDUX_LOGGING` to `VERBOSE` or `VERY_VERBOSE`-- e.g.:

`REDUX_LOGGING=VERBOSE npm run start`
or
`REDUX_LOGGING=VERBOSE npm run simple-serve`

i think i need to update the docs for this... :-(

**Comment by robotblake** [comment]
I don't see anything I'd be concerned about off the top of my head.

**Comment by bcolloran** [comment]
(oh also: there aren't really an JS changes yet in this -- i expect you have other stuff coming in follow-up)

**Comment by wlach** [comment]
@bcolloran the port issue comes from the fact that currently webpack is picking up some stuff from .env, where we had specified these values with our dockerflow-based environment in mind. Oops. For now I'll address this by just serving from port 8000 again, as we do for the internal server -- we can change this later after I move to the next phase of this work, which is to stop specifying server names inside our webpack-generated js altogether.

File: docs/local-dev.md (hunk 1 above)
Comment (wlach) [reply docs/local-dev.md]: It seems like this is the only place where `--reduxVerbose` is mentioned, so I think we can just change it here and be done with it.

File: QA-CHECKLIST.md [hunk 2]
```diff
@@ -11,7 +11,7 @@
 When going through the following checklist, keep the browser console open. There should not be any errors in the console!
 
 a rigourous check will verify that the items listed below work as expected in:
-1. localhost in client only mode (`npm start-and-serve`)
+1. localhost in client only mode (`npm simple-serve`)
```
Comment (rafrombrc) [inline QA-CHECKLIST.md]: This is `npm run simple-serve` everywhere else... should it be here?

Comment (wlach) [reply QA-CHECKLIST.md]: Oops, yes! Good check, will fix before landing

**Pull request merged**